src/process_utility.c

```diff
@@ -1366,9 +1378,28 @@ process_altertable_end_subcmd(Hypertable *ht, Node *parsetree, ObjectAddress *ob
 			}
 			break;
 		case AT_AlterColumnType:
+			if (NULL == ht)
+				break;
+
 			Assert(IsA(cmd->def, ColumnDef));
 			process_alter_column_type_end(ht, cmd);
 			break;
+#if PG10
+		case AT_AttachPartition:
```

Comment:
So, we typically do blocking before regular processUtility, which in this case would be in `process_altertable_start_table()`. Any reason to not do it there? I think that would also avoid the need to pass on a `NULL` hypertable pointer to this function, which generally should deal with alter table handling (recursing) for hypertables after regular postgres handling.

Reply:
Good call - updated